### PR TITLE
chore: add CI

### PR DESCRIPTION
**Pull request by dtopuzov**
- CI with GitHub Actions
- Component tests with https://testing-library.com/

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: "6.x.x"
+
+      - name: Generate Certificates
+        run: dotnet dev-certs https --export-path ./cert/cert.crt --no-password --format PEM
+
+      - name: Install Modules
+        run: |
+          npm i ./framework/build/sitefinity-react-framework-0.1.0.tgz
```
Comment (dechoD) [inline .github/workflows/ci.yml:30]: Is it possible to publish this to a npm feed? Keeping it in the source doesn't feel right.

File: package.json [hunk 2]
```diff
@@ -14,13 +14,21 @@
     "next": "13.4.19",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "typescript": "5.2.2",
-    "sitefinity-react-framework": "file:./framework/build/sitefinity-react-framework-0.1.0.tgz"
+    "sitefinity-react-framework": "file:framework/build/sitefinity-react-framework-0.1.0.tgz",
```
Comment (dechoD) [inline package.json:17]: With this here do we need the explicit install in the ci step?

File: package.json [hunk 3]
```diff
@@ -14,13 +14,21 @@
...
   "scripts": {
     "prod": "next start",
     "dev": "node server.js",
     "build": "next build",
-    "lint": "next lint"
+    "lint": "next lint",
+    "test": "jest --ci"
+  },
+  "devDependencies": {
+    "@testing-library/jest-dom": "^6.1.4",
```
Comment (dechoD) [inline package.json:28]: We usually pin the versions of all our dependencies, please consider it here as well.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          npm i
```
[the remark this answers is not in the document]
Comment (dtopuzov) [reply .github/workflows/ci.yml]: Solved, not project structure is different and we do not install tgz files.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Test
+        run: npm run test
+
+      - name: Build
```
[the remark this answers is not in the document]
Comment (dtopuzov) [reply .github/workflows/ci.yml]: Because build fails :)
Once we fix it it will go before tests.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dtopuzov) [reply .github/workflows/ci.yml]: No need for feeds, with new structure we do not have tgz files in repo.

File: package.json (hunk 2 above)
Comment (dtopuzov) [reply package.json]: no more file:framework/build/sitefinity-react-framework-0.1.0.tgz

File: package.json (hunk 3 above)
Comment (dtopuzov) [reply package.json]: Fixed.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        run: ENVIRONMENT=production && npm run build
```
[the remark this answers is not in the document]
Comment (dtopuzov) [reply .github/workflows/ci.yml]: Changed with `env`.

**Review by dechoD — APPROVED** [review]

File: tests/content-block.test.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
...G93ZWQiOnRydWUsIk1vdmVBbGxvd2VkIjp0cnVlLCJMYWJlbFRvb2x0aXAiOiIiLCJJc1BlcnNvbmFsaXplZCI6ZmFsc2UsIldpZGdldFNlZ21lbnRJZCI6bnVsbCwiV2lkZ2V0U3RhdGUiOlt7IktleSI6ImYwN2Y0ZTZhLWI4MjctNDhmMC1iNmEyLTEzMDVjODBkOWFiYiIsIk5hbWUiOiJTaXRlZmluaXR5Q29udGVudEJsb2NrIiwiQWRkQWxsb3dlZCI6dHJ1ZSwiRGVsZXRlQWxsb3dlZCI6dHJ1ZSwiRWRpdEFsbG93ZWQiOnRydWUsIk1vdmVBbGxvd2VkIjp0cnVlLCJMYWJlbFRvb2x0aXAiOiIiLCJJc1BlcnNvbmFsaXplZCI6ZmFsc2UsIldpZGdldFNlZ21lbnRJZCI6bnVsbH1dLCJQcm9wZXJ0aWVzIjp7IkNvbnRlbnQiOiI8aDEgc3R5bGU9XCJ0ZXh0LWFsaWduOmNlbnRlcjtcIj5UaXRsZTwvaDE+PHA+Jm5ic3A7PC9wPjxwPkZpcnN0IHBhcmFncmFwaC48L3A+PHA+PHN0cm9uZz5Cb2xkIHRlc3Q8L3N0cm9uZz48L3A+PHA+PHN0cm9uZz48L3N0cm9uZz48c3Ryb25nPjxhIGhyZWY9XCJodHRwczovL3d3dy5wcm9ncmVzcy5jb20vXCIgdGl0bGU9XCJUaGlzIGlzIGxpbmsgdG8gcHJvZ3Jlc3MuY29tXCIgZGF0YS1zZi1lYy1pbW11dGFibGU9XCJcIj5odHRwczovL3d3dy5wcm9ncmVzcy5jb20vPC9hPjwvc3Ryb25nPjwvcD48cD48c3Ryb25nPjwvc3Ryb25nPjxpbWcgc3JjPVwiL2ltYWdlcy9kZWZhdWx0LXNvdXJjZS9iYWNrZW5kLWltYWdlcy9zdGFuZGFyZC1jb2xsZWN0aW9uLTEucG5nP3NmdnJzbj1iZTg0MzhmNF8xXCIgYWx0PVwiXCIgc2Ytc2l6ZT1cIjIxOTFcIiBzZnJlZj1cIltpbWFnZXMlN0NPcGVuQWNjZXNzRGF0YVByb3ZpZGVyXTY3MWE3Njk3LTZlOWUtNGUyMi1iZDQ1LWI3M2ZjYzFkZDk0MVwiIC8+PC9wPjxwPiZuYnNwOzwvcD4ifSwiQ2hpbGRyZW4iOltdfQ=='
```
[the remark this answers is not in the document]
Comment (dtopuzov) [reply tests/content-block.test.tsx]: Thanks, should be resolved now.

**Review by zdravkov — APPROVED** [review]

**Pull request merged**